pkg/util/duration/duration.go

```diff
@@ -442,6 +442,67 @@ func (d Duration) Format(buf *bytes.Buffer) {
 	}
 }
 
+// FormatISO8601 emits an ISO 6801 string representation of a Duration to a Buffer truncated to microseconds.
+func (d Duration) FormatISO8601(buf *bytes.Buffer) {
+	buf.WriteString("P")
+
+	if d.nanos < nanosInMicro && d.Days == 0 && d.Months == 0 {
+		buf.WriteString("0D")
+		return
+	}
+
+	if absGE(d.Months, 12) {
+		years := d.Months / 12
+		fmt.Fprintf(buf, "%dY", years)
+		d.Months %= 12
```

Comment:
super nit: prefer assigning months to a variable and modding that. 
paranoid of `d Duration` becoming `d *Duration` and a subtle bug coming later along.